### ISSUE TITLE
Externs: support "declare class" followed by "declare namespace"

### ISSUE THREAD

**Issue by martine** [angular/sickle#107]
es6-promise does this:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/es6-promise/es6-promise.d.ts

Note that they must be in this order; the compiler enforces it.

That means to emit we just have to avoid stepping on the class when emitting the module.

**Comment by martine** [comment]
```js
declare class Foo {}
declare namespace Foo {
  var bar = 3;
}
```

becomes
```js
class Foo {}
var Foo = {};
Foo.bar = 3;
```

we should instead note that we already have a name "Foo" (the class) and drop the `var Foo = {}` part.

**Comment by mprobst** [comment]
Fixed in f5d29aaa1d47a434882c98846a19ee2075fadaa9.
